models/license_plate_detection_yunet/lpd_yunet.py

```diff
@@ -61,10 +61,17 @@ def infer(self, image):
     def _postprocess(self, blob):
         # Decode
         dets = self._decode(blob)
+        
+        # convert to x, y, w, h format
+        bboxes = np.zeros((dets.shape[0], 4))
+        bboxes[:, 0] = np.min(dets[:, [0, 2, 4, 6]], axis=1)  # top-left x
+        bboxes[:, 1] = np.min(dets[:, [1, 3, 5, 7]], axis=1)  # top-left y
+        bboxes[:, 2] = np.max(dets[:, [0, 2, 4, 6]], axis=1) - bboxes[:, 0]  # width
+        bboxes[:, 3] = np.max(dets[:, [1, 3, 5, 7]], axis=1) - bboxes[:, 1]  # height
```

Comment:
Actually `NMSBoxes` supports `RotatedRect` (See https://docs.opencv.org/4.x/d6/d0f/group__dnn.html#gaeec27cb32195e71e6d88032bda193162). So there is no need to de-rotate the boxes.

`self._decode` returns a set of four corners points but `RotatedRect` only needs three points for constructor (See https://docs.opencv.org/4.x/db/dd6/classcv_1_1RotatedRect.html). So try to remove one set of a corner point from `dets` and make it being initialized as `RotatedRect`.

Reply:
If I use something like this to get the first three (x,y) pairs-
```python
rotated_rects = [
    cv.RotatedRect(tuple(det[0:2]), tuple(det[2:4]), tuple(det[4:6])) for det in dets
]
```
I get this error -
```
cv2.error: OpenCV(4.10.0) /io/opencv/modules/core/src/types.cpp:156: error: (-215:Assertion failed) std::fabs(vecs[0].ddot(vecs[1])) * a <= FLT_EPSILON * 9 * x * (norm(vecs[0]) * norm(vecs[1])) in function 'RotatedRect'


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/soham/Documents/Projects/opencv_zoo/models/license_plate_detection_yunet/demo.py", line 121, in <module>
    results = model.infer(frame) # results is a tuple
              ^^^^^^^^^^^^^^^^^^
  File "/home/soham/Documents/Projects/opencv_zoo/models/license_plate_detection_yunet/lpd_yunet.py", line 57, in infer
    results = self._postprocess(outputBlob)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/soham/Documents/Projects/opencv_zoo/models/license_plate_detection_yunet/lpd_yunet.py", line 66, in _postprocess
    cv.RotatedRect(tuple(det[0:2]), tuple(det[2:4]), tuple(det[4:6])) for det in dets
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
SystemError: <class 'cv2.RotatedRect'> returned a result with an exception set
```

Reply:
Hey, I've tried implementing this but am facing this error - 
```
cv2.error: OpenCV(4.10.0) /io/opencv/modules/core/src/types.cpp:156: error: (-215:Assertion failed) std::fabs(vecs[0].ddot(vecs[1])) * a <= FLT_EPSILON * 9 * x * (norm(vecs[0]) * norm(vecs[1])) in function 'RotatedRect'


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/soham/Documents/Projects/opencv_zoo/models/license_plate_detection_yunet/demo.py", line 121, in <module>
    results = model.infer(frame) # results is a tuple
              ^^^^^^^^^^^^^^^^^^
  File "/home/soham/Documents/Projects/opencv_zoo/models/license_plate_detection_yunet/lpd_yunet.py", line 58, in infer
    results = self._postprocess(outputBlob)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/soham/Documents/Projects/opencv_zoo/models/license_plate_detection_yunet/lpd_yunet.py", line 67, in _postprocess
    cv.RotatedRect(tuple(det[0:2]), tuple(det[2:4]), tuple(det[4:6])) for det in dets
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
SystemError: <class 'cv2.RotatedRect'> returned a result with an exception set
```
This is what I'm doing
```python
rotated_rects = [
    cv.RotatedRect(tuple(det[0:2]), tuple(det[2:4]), tuple(det[4:6])) for det in dets
]
```

Reply:
@souhhmm Do you have updates on this?

Reply:
Hey, yea although in vain. I've been encountering this error while trying to implement this.
```
cv2.error: OpenCV(4.10.0) /io/opencv/modules/core/src/types.cpp:156: error: (-215:Assertion failed) std::fabs(vecs[0].ddot(vecs[1])) * a <= FLT_EPSILON * 9 * x * (norm(vecs[0]) * norm(vecs[1])) in function 'RotatedRect'


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/media/soham/New Volume/Projects 2/opencv_zoo/models/license_plate_detection_yunet/demo.py", line 121, in <module>
    results = model.infer(frame) # results is a tuple
              ^^^^^^^^^^^^^^^^^^
  File "/media/soham/New Volume/Projects 2/opencv_zoo/models/license_plate_detection_yunet/lpd_yunet.py", line 57, in infer
    results = self._postprocess(outputBlob)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/media/soham/New Volume/Projects 2/opencv_zoo/models/license_plate_detection_yunet/lpd_yunet.py", line 66, in _postprocess
    cv.RotatedRect(tuple(det[0:2]), tuple(det[2:4]), tuple(det[4:6])) for det in dets
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
SystemError: <class 'cv2.RotatedRect'> returned a result with an exception set
```

Reply:
@fengyuentau Do you have any feedback on where I might be going wrong?